heron/spi/src/java/com/twitter/heron/spi/statemgr/IStateManager.java

```diff
@@ -228,52 +248,56 @@ public String getName() {
       WatchCallback watcher, String topologyName);
 
   /**
-   * Get the packing plan for the given topology
+   * Delete the execution state for the given topology
    *
-   * @param watcher @see com.twitter.heron.spi.statemgr.WatchCallback
-   * @return PackingPlan
+   * @return Boolean - Success or Failure
    */
-  ListenableFuture<PackingPlans.PackingPlan> getPackingPlan(
-      WatchCallback watcher, String topologyName);
+  ListenableFuture<Boolean> deleteExecutionState(String topologyName);
 
   /**
-   * Set the location of Tmaster.
+   * Set the scheduler location for the given topology
    *
    * @return Boolean - Success or Failure
    */
-  ListenableFuture<Boolean> setTMasterLocation(
-      TopologyMaster.TMasterLocation location, String topologyName);
+  ListenableFuture<Boolean> setSchedulerLocation(
+      Scheduler.SchedulerLocation location, String topologyName);
 
   /**
-   * Set the location of MetricsCache.
+   * Get the scheduler location for the given topology
    *
-   * @return Boolean - Success or Failure
+   * @param watcher @see com.twitter.heron.spi.statemgr.WatchCallback
+   * @return SchedulerLocation
    */
-  ListenableFuture<Boolean> setMetricsCacheLocation(
-      TopologyMaster.MetricsCacheLocation location, String topologyName);
+  ListenableFuture<Scheduler.SchedulerLocation> getSchedulerLocation(
+      WatchCallback watcher, String topologyName);
 
   /**
-   * Set the physical plan for the given topology
+   * Delete the scheduler location for the given topology
    *
    * @return Boolean - Success or Failure
    */
-  ListenableFuture<Boolean> setPhysicalPlan(
-      PhysicalPlans.PhysicalPlan physicalPlan, String topologyName);
+  ListenableFuture<Boolean> deleteSchedulerLocation(String topologyName);
 
   /**
-   * Set the packing plan for the given topology
+   * Set the Stateful Checkpoint
    *
    * @return Boolean - Success or Failure
    */
-  ListenableFuture<Boolean> setPackingPlan(
-      PackingPlans.PackingPlan packingPlan, String topologyName);
+  ListenableFuture<Boolean> setStatefulCheckpoint(
+      CheckpointManager.StatefulConsistentCheckpoints checkpoint, String topologyName);
 
   /**
-   * Get the physical plan for the given topology
+   * Get the Stateful Checkpoint
    *
-   * @param watcher @see com.twitter.heron.spi.statemgr.WatchCallback
-   * @return PhysicalPlan
+   * @return Boolean - Success or Failure
    */
-  ListenableFuture<PhysicalPlans.PhysicalPlan> getPhysicalPlan(
+  ListenableFuture<CheckpointManager.StatefulConsistentCheckpoints> getStatefulCheckpoints(
       WatchCallback watcher, String topologyName);
+
+  /**
+   * Delete the stateful checkpoint for the given topology
+   *
+   * @return Boolean - Success or Failure
+   */
+  ListenableFuture<Boolean> deleteStatefulCheckpoint(String topologyName);
```

Comment:
The deleteStatefulCheckpoint method needs to take an identifier to make it clear which checkpoint is to be deleted. The state server stores multiple checkpoints per topology